backend/app/assets/javascripts/spree/backend/admin.js.erb

```diff
@@ -111,8 +111,9 @@ $(document).ready(function(){
     new_table_row.find("input, select").each(function () {
       var el = $(this);
       el.val("");
-      el.prop("id", el.prop("id").replace(/\d+/, new_id))
-      el.prop("name", el.prop("name").replace(/\d+/, new_id))
+      // Replace last occurrence of a number
+      el.prop("id", el.prop("id").replace(/\d+(?=[^\d]*$)/, new_id))
+      el.prop("name", el.prop("name").replace(/\d+(?=[^\d]*$)/, new_id))
```

Comment:
Here's an example of a generated name property for a value input:
`product[variant_property_rules_attributes][0][values_attributes][0][property_name]`

With the previous code, when creating a new input, the generated name would be something like this:
`product[variant_property_rules_attributes][1442601174172][values_attributes][1442601174172][property_name]`

We want all of the values to be associated with the same variant_property_rule for when we call `update_attributes` so this code generates `product[variant_property_rules_attributes][0][values_attributes][1442601174172][property_name]` instead by only replacing the last numeric value in the string.